ocaml/xcp-rrdd/bin/rrdd/rrdd_server.ml

```diff
@@ -456,6 +457,30 @@ let query_host_ds (ds_name : string) : float =
           query ds_name rrdi
   )
 
+(** Dump all latest data of host dss to file in json format so that any client
+    can read even if it's non-privileged user, such as NRPE. *)
+let dump_host_dss_to_file (file : string) : unit =
+  let json =
+    with_lock mutex (fun () ->
+        match !host_rrd with
+        | None ->
+            `Assoc []
+        | Some rrdi ->
+            `Assoc
+              (Rrd.ds_names rrdi.rrd
+              |> List.map (fun ds_name -> (ds_name, `Float (query ds_name rrdi)))
+              )
+    )
+    |> Yojson.Basic.to_string
+    |> Bytes.of_string
+  in
+  let fd =
```

Comment:
`Unixext.write_string_to_file` could be used. But would be better to use the Yojson functions that write to a channel rather than creating a potentially large string first.

Reply:
Thanks! fixed now.